### PR TITLE
Red Black Tree C++ code added ---> Fixes #1679

### LINKED ISSUE #1679: Red Black Tree
A **red-black tree** is a self-balancing binary search tree where each node has an extra bit, and that bit is often interpreted as the colour (red or black).
It reduces the searching time to O(log n), where n is the total number of elements in the tree. 

I want to implement **Insertion and Deletion** in Red Black Tree using **C++**.

Please assign me this issue.[](url)

### PR DESCRIPTION
**Pull request by mehak6569**
This code depicts the insertion and deletion in Red Black Tree using C++.

Fixes #1679

## Checklist:

- [x] My code follows the guidelines of this project
- [x] I have tested my code and it works properly
- [x] My changes generate no new warnings
- [x] This is my original code and I have NOT copied the code from anywhere else.

### CODE REVIEW

**Comment by mehak6569** [comment]
@tarun26091999 @plazzy99 @rudrakshi99 Please review my PR.

**Comment by mehak6569** [comment]
@rish-singhal Please review and merge my PR

File: Tree/readme.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ that, this property should be satisfied at every node in the tree.
 * Level Order Traversal in BST ----> [C++]() | [Java]() | [Python](/Code/Python/level_order_traversal_binary_tree.py)
 * Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp)
 * Lowest common ancestor in a binary tree ----> [Python](/Code/Python/LCA_in_binary_tree.py)
+* Red Black Tree (Insertion and Deletion operation) ----> [C++](/Code/C++/RedBlack_tree.cpp) | [Java](/Code/Java/RedBlackTree.java)  
 * Right Side View of Binary Tree ----> [C++](/Code/C++/Right_Side_View_of_Binary_Tree.cpp)
```
Comment (rish-singhal) [inline Tree/readme.md:67]: Sort this list

File: Code/C++/RedBlack_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+			inOrder(node->right);
+		} 
+	}
+
+	void postOrder(NodePtr node) {
+		if (node != TNULL) {
+			postOrder(node->left);
+			postOrder(node->right);
+			string node_Color = node->color?"RED":"BLACK";
+		   cout<<node->data<<"("<<node_Color<<")  ";
```
Comment (rish-singhal) [inline Code/C++/RedBlack_tree.cpp:62]: Give spaces around <<,>>,... everywhere also around operators

File: Code/C++/RedBlack_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+        case 4:
+            break;   
+        }
+
+    } while(choice != 4);
+
+	return 0;
+}
+
+/*
```
Comment (rish-singhal) [inline Code/C++/RedBlack_tree.cpp:493]: Provide atleast 2 test cases and complexity of each function

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Code/C++/RedBlack_tree.cpp (hunk 2 above)
Comment (rish-singhal) [reply Code/C++/RedBlack_tree.cpp:62]: Update.

File: Tree/readme.md (hunk 1 above)
Comment (rish-singhal) [reply Tree/readme.md:67]: Update this

Comment (mehak6569) [reply Tree/readme.md:67]: I have sorted the whole list. Is it correct?

**Pull request merged**